spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/ImageTransferLearning.scala

```diff
@@ -0,0 +1,155 @@
+import com.intel.analytics.bigdl.transform.vision.image.augmentation.{CenterCrop, ChannelNormalize, Resize}
+import com.intel.analytics.bigdl.utils.Engine
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.ml.adapter.{HasFeaturesCol, HasPredictionCol}
+import org.apache.spark.ml.classification.{LogisticRegression, LogisticRegressionModel}
+import org.apache.spark.ml.evaluation.MulticlassClassificationEvaluator
+import org.apache.spark.ml.linalg.SQLDataTypes.VectorType
+import org.apache.spark.ml.linalg.{Vector, Vectors}
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.{Pipeline, Transformer}
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Dataset, SQLContext, SparkSession}
+import scopt.OptionParser
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+
+
+object TransferLearning {
+
+  def main(args: Array[String]): Unit = {
+
+    val defaultParams = Utils.LocalParams()
+
+    Utils.parser.parse(args, defaultParams).map { params =>
+
+      val conf = Engine.createSparkConf().setAppName("TransferLearning")
+      val spark = SparkSession.builder().config(conf).getOrCreate()
+      Engine.init
+
+      val createLabel = udf((name: String) => if (name.contains("cat")) 1.0 else 0.0)
+      val imagesDF: DataFrame = Utils.loadImages(params.folder, params.batchSize, spark.sqlContext)
+        .withColumn("label", createLabel(col("imageName")))
+        .withColumnRenamed("features", "imageFeatures")
+
+      val Array(validationDF, trainingDF) = imagesDF.randomSplit(Array(0.90, 0.10), seed = 1L)
+
+      val criterion = ClassNLLCriterion[Float]()
+
+      val loadedModel: AbstractModule[Activity, Activity, Float] = Module
+        .loadCaffeModel[Float](params.caffeDefPath, params.modelPath)
+
+      val featurizer = new DLModel[Float](loadedModel, Array(3, 224, 224))
+        .setBatchSize(params.batchSize)
+        .setFeaturesCol("imageFeatures")
+        .setPredictionCol("tmp1")
+
+
+      val schemaTransformer = new Array2Vector()
+        .setFeaturesCol("tmp1")
+        .setPredictionCol("features")
+
+      val lr = new LogisticRegression()
+        .setMaxIter(20)
+        .setRegParam(0.05)
+        .setElasticNetParam(0.3)
+        .setFeaturesCol("features")
+
+      val pipeline = new Pipeline().setStages(
+        Array(featurizer, schemaTransformer, lr))
+
+      val pipelineModel = pipeline.fit(trainingDF)
+
+      val predictions = pipelineModel.transform(trainingDF)
+
+      predictions.show(200)
+      predictions.printSchema()
+
+      val evaluation = new MulticlassClassificationEvaluator().setPredictionCol("prediction")
+        .setMetricName("accuracy").evaluate(predictions)
+      println("evaluation result on validationDF: " + evaluation)
+
+    }
+  }
+
+}
+
+class Array2Vector(override val uid: String = "array2vector") extends Transformer with HasFeaturesCol
```

Comment:
can we use a udf for this?

Reply:
I want to use the pipeline then I have to do this this way. If I want to manually connect the steps, this is not necessary